src/Generators/Microsoft.Gen.Metrics/Emitter.cs

```diff
@@ -297,7 +297,7 @@ private void GenTagList(MetricMethod metricMethod)
             foreach (var tagName in metricMethod.TagKeys)
             {
                 var paramName = GetSanitizedParamName(tagName);
-                OutLn($"new global::System.Collections.Generic.KeyValuePair<string, object?>(\"{paramName}\", {paramName}),");
+                OutLn($"new global::System.Collections.Generic.KeyValuePair<string, object?>(\"{tagName}\", {paramName}),");
```

Comment:
FYI @klauco: the generator now emits dots, colons and dashes as-is in dimension names